src/abstractions/xunit.abstractions/Frameworks/ISourceInformationProvider.cs

```diff
@@ -15,5 +15,14 @@ public interface ISourceInformationProvider : IDisposable
         /// <returns>The source information, with null string and int values when the information is not available.
         /// Note: return value should never be <c>null</c>, only the interior data values inside.</returns>
         ISourceInformation GetSourceInformation(ITestCase testCase);
+
+        /// <summary>
+        /// Returns the source information for a test method inside a test class.
+        /// </summary>
+        /// <param name="typeName">Test class containing the method.</param>
+        /// <param name="methodName">The test case to retrieve information for.</param>
+        /// <returns>The source information, with null string and int values when the information is not available.
+        /// Note: return value should never be <c>null</c>, only the interior data values inside.</returns>
+        ISourceInformation GetSourceInformation(string typeName, string methodName);
```

Comment:
Abstractions cannot be changed. Abstractions is our forward and backward compatibility layer for runners.

This is an iron-clad rule. There are no exceptions.